presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java

```diff
@@ -512,6 +513,15 @@ public PlanOptimizers(
         // TODO: consider adding a formal final plan sanitization optimizer that prepares the plan for transmission/execution/logging
         // TODO: figure out how to improve the set flattening optimizer so that it can run at any point
 
+        // TODO: move this before optimization if possible!!
+        // Replace all expressions with row expressions
+        IterativeOptimizer expressionTranslator = new IterativeOptimizer(
```

Comment:
Is this for test purpose? Why do we add this here?

Reply:
This is where the translation from `Expression` -> `RowExpression` happens. No more `RawExpression` after this point.